### PR TITLE
Bring article progress sync through API

### PR DESCRIPTION
**Pull request by tcitworld**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | no
| Documentation | no
| Translation   | no
| Fixed tickets | #2643
| License       | MIT

### CODE REVIEW

File: src/Wallabag/CoreBundle/Entity/Entry.php [hunk 1]
```diff
@@ -680,6 +689,7 @@ public function cleanUuid()
     }
 
     /**
+<<<<<<< Updated upstream
```
Comment (tcitworld) [inline src/Wallabag/CoreBundle/Entity/Entry.php]: Boo

**Comment by nicosomb** [comment]
We also need to add the code in the view of the web application (to send the reading progress, maybe with javascript like @jlnostr wrote)

File: src/Wallabag/ApiBundle/Controller/EntryRestController.php [hunk 2]
```diff
@@ -272,6 +274,13 @@ public function patchEntriesAction(Entry $entry, Request $request)
             $entry->setStarred((bool) $isStarred);
         }
 
+        if (!is_null($progress)) {
```
Comment (nicosomb) [inline src/Wallabag/ApiBundle/Controller/EntryRestController.php]: @j0k3r is that better to use `is_null` or `null ===` ?

File: src/Wallabag/ApiBundle/Controller/EntryRestController.php [hunk 3]
```diff
@@ -272,6 +274,13 @@ public function patchEntriesAction(Entry $entry, Request $request)
...
+            $progress = (int) $progress;
+            if ($progress >= 0 && $progress <= 100) {
+                $entry->setProgress((int) $progress);
```
Comment (nicosomb) [inline src/Wallabag/ApiBundle/Controller/EntryRestController.php]: $entry->setProgress($progress);

File: src/Wallabag/ApiBundle/Controller/EntryRestController.php (hunk 2 above)
Comment (tcitworld) [inline src/Wallabag/ApiBundle/Controller/EntryRestController.php]: http://stackoverflow.com/a/8229005

Comment (j0k3r) [inline src/Wallabag/ApiBundle/Controller/EntryRestController.php]: It seems that we don't often use `is_null`, so if we want to keep consistency, we should use `null ===`

**Comment by tcitworld** [comment]
Rebased.

**Comment by tcitworld** [comment]
Rebased.

**Comment by actuallynotjohndoe** [comment]
Do you plan to implement this feature in the near future? The past week I've been using wallabag quite intensive and to be able to sync the reading progress between devices would be a killer feature! I guess I'm not the only one that would really like this feature.

**Comment by RrOoSsSsOo** [comment]
I would like to start by saying that from my point of view this is a fundamental feature to have in Wallabag and should be implemented as soon as possible.
If it is implemented then a new "**Continue reading**" section should also be prepared in true _Netflix-style_ (and perhaps integrated well into a new homepage)

**Pull request closed without merging**